### PR TITLE
add example: how to stub loading of ressources

### LINKED ISSUE #378: New example: Mock loading of ressources
Usecase:
Based upon any action a ressource (like an image) is loaded. Those requests can not be mocked by `cy.route`. 
Today I added an answer to such a question on stackoverflow:
https://stackoverflow.com/questions/54481634/how-to-mock-an-image-with-a-fixture-in-cypress/59291464#59291464

To make it public to the cypress community I thought it would be nice to have this example in this repo. Please close this issue and the attached PR if you don't think so.

### PR DESCRIPTION
**Pull request by gabbersepp**
add example: how to stub loading of ressources

closes #378

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,6 +76,7 @@ Recipe | Description
 [Stubbing `window.open` and `console.log`](./examples/stubbing-spying__window) | Use `cy.stub()` and `cy.spy()` to test application behavior
 [Stubbing Google Analytics](./examples/stubbing-spying__google-analytics) | Use `cy.stub()` to test Google Analytics calls
 [Spying and stubbing methods on `console` object](./examples/stubbing-spying__console) | Use `cy.spy()` and `cy.stub()` on `console.log`
+[Stub ressource loading](./examples/stubbing__ressources) | Use `MutationObserver` to stub ressource loading like `img` tags
```
Comment (jennifer-shehane) [inline README.md]: Resources is misspelled in a few places, everywhere there is `ressources` should be `resources`

```suggestion
[Stub resource loading](./examples/stubbing__resources) | Use `MutationObserver` to stub resource loading like `img` tags
```

File: examples/stubbing__resources/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Stubbing Ressources
```
Comment (jennifer-shehane) [inline examples/stubbing__resources/README.md]:
```suggestion
# Stubbing Resources
```

File: examples/stubbing__resources/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+The spec file [cypress/integration/spec.js](cypress/integration/spec.js) shows how to stub ressources that are loaded e.g. by `img` tag.
```
Comment (jennifer-shehane) [inline examples/stubbing__resources/README.md]:
```suggestion
The spec file [cypress/integration/spec.js](cypress/integration/spec.js) shows how to stub resources that are loaded e.g. by `img` tag.
```

File: examples/stubbing__resources/cypress/integration/spec.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/// <reference types="cypress" />
+var MutationObserver = window.MutationObserver || window.WebKitMutationObserver;
+
+describe('Stub loading of ressources', () => {
```
Comment (jennifer-shehane) [inline examples/stubbing__resources/cypress/integration/spec.js]:
```suggestion
describe('Stub loading of resources', () => {
```

File: examples/stubbing__resources/package.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "stubbing-ressources",
```
Comment (jennifer-shehane) [inline examples/stubbing__resources/package.json]:
```suggestion
  "name": "stubbing-resources",
```

File: examples/stubbing__resources/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+
+When clicking onto the button in the `index.html`, an api call is simulated that would provide an image url. This image is manipulated in order to display a predefined image.
+
+![Test example](images/ressources-example.png)
```
Comment (jennifer-shehane) [inline examples/stubbing__resources/README.md]: The ressources-example.png will need to be renamed to resources-example.png also

```suggestion
![Test example](images/resources-example.png)
```

**Comment by gabbersepp** [comment]
Thanks! I have written it correctly in ´circly.yml`. Well done :-D

File: examples/stubbing__resources/cypress/integration/spec.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        src: "https://cypress.io/cypress.png"
+      })
+    });
+
+    cy.visit("index.html").then(() => {
+      cy.get("#img-container").then($element => {
+        var el = $element[0];
+        observeDOM(el, mutations => {
+          // red dot
+          mutations[0].addedNodes[0].src = "data:image/png;base64," +
```
Comment (bahmutov) [inline examples/stubbing__resources/cypress/integration/spec.js]: can you load the mocked image from a fixture file?

File: examples/stubbing__resources/index.html [hunk 8]
```diff
@@ -0,0 +1,34 @@
+<html>
+
+<body>
+  <button id="button">load</button>
+  <div id="img-container">
+  </div>
+
+  <script>
+    function get(url) {
```
Comment (bahmutov) [inline examples/stubbing__resources/index.html]: I think having an Ajax call just to stub it makes the example more confusing than it should be. I would be all for observing the image element and switching the image to fixture.

File: examples/stubbing__resources/cypress/integration/spec.js (hunk 7 above)
Comment (gabbersepp) [reply examples/stubbing__resources/cypress/integration/spec.js]: Yes of course. I have added it in the same way I am doing it in my project.

File: examples/stubbing__resources/index.html (hunk 8 above)
Comment (gabbersepp) [reply examples/stubbing__resources/index.html]: You are right. This was not very helpful and does not make sense at all. Now the image tag is added directly by clicking onto the button

**Pull request merged**